[Security Solution][Threat Hunting] Replace heavy accessibility rtl selectors

kapral18 opened the pull request:
addresses #223242

So for this specific PR the gains are almost invisible:

Before
[timing-results-before.csv](https://github.com/user-attachments/files/21034673/timing-results-before.csv)

After:
[timing-results-after.csv](https://github.com/user-attachments/files/21034675/timing-results-after.csv)

Because this PR tests are testing pretty shallow rendering trees or selectors are doing too little work to see the difference.

Whereas results from earlier [PR](https://github.com/elastic/kibana/pull/196591) rendering deep trees and involving some selector sweat shows convincing time savings.

Before:
[timing-results-before.csv](https://github.com/user-attachments/files/21034732/timing-results-before.csv)

After:
[timing-results-after.csv](https://github.com/user-attachments/files/21034733/timing-results-after.csv)

I have raised this issue last week on one of our team meetings and asked about this discrepancy and whether it's ok to still move on with banning heavy selectors for us as a team. And the team decided that it's ok as long as we potentially save time on complex trees.

To get an idea of the algorithmic scaling of the cost with deeper trees horizontally or vertically I have created a perf-test repo where I go over each selector in depth and test them out in various ways to try to highlight said difference and impact.

Please feel free to explore this if you are curious and when you have time: https://github.com/elastic/rtl-selectors-perf-test

kapral18 committed:
[Security Solution][Threat Hunting] Replace heavy accessibility rtl selectors

addresses #223242

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308115)
* Commit: 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #7](https://buildkite.com/elastic/kibana-pull-request/builds/308115#01976630-f52e-4638-9be5-127aad8b7257)

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308115","number":308115,"commit":"20a3dfd5608612e36a379a5527ae21e3b2e20b4c"}],"number":308115}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308577)
* Commit: faf050f745f7caa2a37433c9f1c81de50030c62d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/22e6f78b843d8fc6a4504e3baae065340323f882) | [after](https://github.com/elastic/kibana/commit/faf050f745f7caa2a37433c9f1c81de50030c62d) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.3MB | 9.3MB | +142.0B |



### History
* :broken_heart: [Build #308115](https://buildkite.com/elastic/kibana-pull-request/builds/308115) failed 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308577","number":308577,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308115","number":308115,"commit":"20a3dfd5608612e36a379a5527ae21e3b2e20b4c"}],"number":308577}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312148)
* Commit: faf050f745f7caa2a37433c9f1c81de50030c62d

### Failed CI Steps
* [x-pack/test/alerting_api_integration/security_and_spaces/group1/config.ts](https://buildkite.com/elastic/kibana-pull-request/builds/312148#0197a27f-65fb-4b7c-9008-4c628d2fe6c7)
* [Detection Engine - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/312148#0197a263-5655-463e-9ca1-353a398fa275)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312148#0197a27f-65fb-4b7c-9008-4c628d2fe6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312148/jobs/0197a27f-65fb-4b7c-9008-4c628d2fe6c7/artifacts/0197a28f-c9f1-4b3e-be6f-3af7764b3bea) x-pack/test/alerting_api_integration/security_and_spaces/group1/config.ts / alerting api integration security and spaces enabled Alerts - Group 1 alerts backfill rule runs ad hoc backfill task should run all execution sets of a scheduled backfill and correctly generate alerts

### History
* :green_heart: [Build #308577](https://buildkite.com/elastic/kibana-pull-request/builds/308577) succeeded faf050f745f7caa2a37433c9f1c81de50030c62d
* :broken_heart: [Build #308115](https://buildkite.com/elastic/kibana-pull-request/builds/308115) failed 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312148","number":312148,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308577","number":308577,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308115","number":308115,"commit":"20a3dfd5608612e36a379a5527ae21e3b2e20b4c"}],"number":312148}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

jughosta reviewed (approved):
Changes in Data discovery tests LGTM, thanks!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/314817)
* Commit: 846d6b41b830901d1f1c2e19ec9f3451feff8d13

### Failed CI Steps
* [Osquery Cypress Tests on Serverless #8](https://buildkite.com/elastic/kibana-pull-request/builds/314817#0197caa4-718a-4b5d-9334-9539f0eb5a22)

### History
* :broken_heart: [Build #312148](https://buildkite.com/elastic/kibana-pull-request/builds/312148) failed faf050f745f7caa2a37433c9f1c81de50030c62d
* :green_heart: [Build #308577](https://buildkite.com/elastic/kibana-pull-request/builds/308577) succeeded faf050f745f7caa2a37433c9f1c81de50030c62d
* :broken_heart: [Build #308115](https://buildkite.com/elastic/kibana-pull-request/builds/308115) failed 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314817","number":314817,"commit":"846d6b41b830901d1f1c2e19ec9f3451feff8d13"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312148","number":312148,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308577","number":308577,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308115","number":308115,"commit":"20a3dfd5608612e36a379a5527ae21e3b2e20b4c"}],"number":314817}
buildkite-pr-comment-->

janmonschke commented:
@kapral18 Would it be possible to get some sort of number on runtime improvements (even if it's just locally on your machine)?

logeekal commented on src/platform/packages/shared/kbn-cell-actions/.eslintrc.js:
```diff
@@ -73,11 +73,30 @@ const path = require('path');
 
 /** @type {Array.<RestrictedImportPath>} */
-const RESTRICTED_IMPORTS = [
+const RESTRICTED_IMPORTS_PATHS = [
```
@kapral18 , is there a way to DRY this script to avoid repeating the almost same script in each `.eslintrc` does

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/314913)
* Commit: ed053abba60f7c332bad8a2c4424207fa254ede8

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #12](https://buildkite.com/elastic/kibana-pull-request/builds/314913#0197cb2a-457c-4f19-bc87-c2af8fc2be7e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e263d5087115a3fe3644391bf01c0cf1c9aa71a9) | [after](https://github.com/elastic/kibana/commit/ed053abba60f7c332bad8a2c4424207fa254ede8) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.8MB | 9.8MB | +110.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/e263d5087115a3fe3644391bf01c0cf1c9aa71a9) | [after](https://github.com/elastic/kibana/commit/ed053abba60f7c332bad8a2c4424207fa254ede8) | diff |
| --- | --- | --- | --- |
| `@kbn/securitysolution-data-table` | 3 | 5 | +2 |
| `@kbn/unified-data-table` | 10 | 12 | +2 |
| `securitySolution` | 629 | 657 | +28 |
| `timelines` | 24 | 26 | +2 |
| total |  |  | +34 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/e263d5087115a3fe3644391bf01c0cf1c9aa71a9) | [after](https://github.com/elastic/kibana/commit/ed053abba60f7c332bad8a2c4424207fa254ede8) | diff |
| --- | --- | --- | --- |
| `@kbn/securitysolution-data-table` | 3 | 5 | +2 |
| `@kbn/unified-data-table` | 10 | 12 | +2 |
| `securitySolution` | 727 | 755 | +28 |
| `timelines` | 24 | 26 | +2 |
| total |  |  | +34 |


</details>


### History
* :broken_heart: [Build #314817](https://buildkite.com/elastic/kibana-pull-request/builds/314817) failed 846d6b41b830901d1f1c2e19ec9f3451feff8d13
* :broken_heart: [Build #312148](https://buildkite.com/elastic/kibana-pull-request/builds/312148) failed faf050f745f7caa2a37433c9f1c81de50030c62d
* :green_heart: [Build #308577](https://buildkite.com/elastic/kibana-pull-request/builds/308577) succeeded faf050f745f7caa2a37433c9f1c81de50030c62d
* :broken_heart: [Build #308115](https://buildkite.com/elastic/kibana-pull-request/builds/308115) failed 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314913","number":314913,"commit":"ed053abba60f7c332bad8a2c4424207fa254ede8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314817","number":314817,"commit":"846d6b41b830901d1f1c2e19ec9f3451feff8d13"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312148","number":312148,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308577","number":308577,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308115","number":308115,"commit":"20a3dfd5608612e36a379a5527ae21e3b2e20b4c"}],"number":314913}
buildkite-pr-comment-->

kapral18 commented:
> @kapral18 Would it be possible to get some sort of number on runtime improvements (even if it's just locally on your machine)?

I updated the description of the PR @janmonschke @jughosta @logeekal

kapral18 commented:
/ci

kapral18 commented:
> @kapral18 Thanks for the detailed investigation on this, looks like the impact here can be pretty significant, although with the tradeoff of reduced a11y coverage. I wonder if there should be a broader communication about this outside of the Security team since it's probably unexpected/unintuitive to most considering it goes against RTL best practices?

Hi @davismcphee thanks for the interest. I actually lead this effort for a long time now. I started the heavy selector assessment as an ADR and involved a lot of folks outside of security including appex and eui letting them know about the potential impact. It's an ongoing issue, there is a google doc (ADR) that has been thoroughly discussed and a related channel that I created about this. I also created a couple of tools to aid in assessing this using gcp infra. I am still finalizing those.

After I finish those I will post all my findings and the tools to verify them again to the bigger group and I will send you an invite as well into the channel and discussion if you are interested.

davismcphee commented:
@kapral18 Thanks the context and your investigations here! Looking forward to it, and would definitely appreciate an invite to the channel too.

PhilippeOberti reviewed (commented):
This is a huge amount of work, thank you so much for doing that.
Everything looks great, but I do have one small request related to the `.eslintrc.json` files, especially the few in the `detections` folder.
- `public/detections/components/alert_summary/.eslintrc.js`
- `public/detections/components/alerts_kpis/.eslintrc.js`
- `public/detections/components/alerts_table/.eslintrc.js`

Could we have a single one at the root of the `detections` folder instead?
The reason I'm asking is because I'm making drastic changes to the content of that folder (draft [PR](https://github.com/elastic/kibana/pull/222457) here) so having it at the root would prevent me from remembering to add/move them.
Plus it would be more future proof...

I'm happy to take a pass and help fixing the other tests if needed!

Or we could just ignore this folder for now to merge this already huge achievement you have here!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/332416)
* Commit: 832ad4210b346abc9724585bdbeca03f70fe7e2e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d354cd945329081065c130e26bd103821f4d32fb) | [after](https://github.com/elastic/kibana/commit/832ad4210b346abc9724585bdbeca03f70fe7e2e) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.4MB | 10.4MB | +110.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/d354cd945329081065c130e26bd103821f4d32fb) | [after](https://github.com/elastic/kibana/commit/832ad4210b346abc9724585bdbeca03f70fe7e2e) | diff |
| --- | --- | --- | --- |
| `@kbn/securitysolution-data-table` | 3 | 5 | +2 |
| `@kbn/unified-data-table` | 10 | 12 | +2 |
| `securitySolution` | 680 | 708 | +28 |
| `timelines` | 24 | 26 | +2 |
| total |  |  | +34 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/d354cd945329081065c130e26bd103821f4d32fb) | [after](https://github.com/elastic/kibana/commit/832ad4210b346abc9724585bdbeca03f70fe7e2e) | diff |
| --- | --- | --- | --- |
| `@kbn/securitysolution-data-table` | 3 | 5 | +2 |
| `@kbn/unified-data-table` | 10 | 12 | +2 |
| `securitySolution` | 779 | 807 | +28 |
| `timelines` | 24 | 26 | +2 |
| total |  |  | +34 |


</details>


### History
* :green_heart: [Build #316260](https://buildkite.com/elastic/kibana-pull-request/builds/316260) succeeded ef53fc914eefde7b44514a483817fb36e725e2da
* :yellow_heart: [Build #314913](https://buildkite.com/elastic/kibana-pull-request/builds/314913) was flaky ed053abba60f7c332bad8a2c4424207fa254ede8
* :broken_heart: [Build #314817](https://buildkite.com/elastic/kibana-pull-request/builds/314817) failed 846d6b41b830901d1f1c2e19ec9f3451feff8d13
* :broken_heart: [Build #312148](https://buildkite.com/elastic/kibana-pull-request/builds/312148) failed faf050f745f7caa2a37433c9f1c81de50030c62d
* :green_heart: [Build #308577](https://buildkite.com/elastic/kibana-pull-request/builds/308577) succeeded faf050f745f7caa2a37433c9f1c81de50030c62d
* :broken_heart: [Build #308115](https://buildkite.com/elastic/kibana-pull-request/builds/308115) failed 20a3dfd5608612e36a379a5527ae21e3b2e20b4c

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332416","number":332416,"commit":"832ad4210b346abc9724585bdbeca03f70fe7e2e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/316260","number":316260,"commit":"ef53fc914eefde7b44514a483817fb36e725e2da"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314913","number":314913,"commit":"ed053abba60f7c332bad8a2c4424207fa254ede8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314817","number":314817,"commit":"846d6b41b830901d1f1c2e19ec9f3451feff8d13"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312148","number":312148,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308577","number":308577,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"}],"number":332416}
buildkite-pr-comment-->

kapral18 commented:
> This is a huge amount of work, thank you so much for doing that. Everything looks great, but I do have one small request related to the `.eslintrc.json` files, especially the few in the `detections` folder.
> 
> * `public/detections/components/alert_summary/.eslintrc.js`
> * `public/detections/components/alerts_kpis/.eslintrc.js`
> * `public/detections/components/alerts_table/.eslintrc.js`
> 
> Could we have a single one at the root of the `detections` folder instead? The reason I'm asking is because I'm making drastic changes to the content of that folder (draft [PR](https://github.com/elastic/kibana/pull/222457) here) so having it at the root would prevent me from remembering to add/move them. Plus it would be more future proof...
> 
> I'm happy to take a pass and help fixing the other tests if needed!
> 
> Or we could just ignore this folder for now to merge this already huge achievement you have here!

Thanks for the kind words @PhilippeOberti. 

The reason I added `.eslintrc.js` in subdirectories was that not all `detections` folders belong to us. So I didn't want to enforce our linting rules in folders of other teams. But if after your PR we will own detections then I can move it to root of detections. Let me know.

kapral18 committed:
fix: simplify and standardize ESLint configuration imports

- Removed dynamic root directory calculation and replaced with static relative paths.
- Replaced `JSON.parse(JSON.stringify(rootConfig))` with `structuredClone(rootConfig)` for deep cloning.
- Removed unused `execSync` imports.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/332565)
* Commit: 6d550e2a4e58de08834f24153a2eefaf74f162f2

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/332565#0198e5d0-332b-4722-8592-92c86ef33088)

### History
* :green_heart: [Build #332416](https://buildkite.com/elastic/kibana-pull-request/builds/332416) succeeded 832ad4210b346abc9724585bdbeca03f70fe7e2e
* :green_heart: [Build #316260](https://buildkite.com/elastic/kibana-pull-request/builds/316260) succeeded ef53fc914eefde7b44514a483817fb36e725e2da
* :yellow_heart: [Build #314913](https://buildkite.com/elastic/kibana-pull-request/builds/314913) was flaky ed053abba60f7c332bad8a2c4424207fa254ede8
* :broken_heart: [Build #314817](https://buildkite.com/elastic/kibana-pull-request/builds/314817) failed 846d6b41b830901d1f1c2e19ec9f3451feff8d13
* :broken_heart: [Build #312148](https://buildkite.com/elastic/kibana-pull-request/builds/312148) failed faf050f745f7caa2a37433c9f1c81de50030c62d
* :green_heart: [Build #308577](https://buildkite.com/elastic/kibana-pull-request/builds/308577) succeeded faf050f745f7caa2a37433c9f1c81de50030c62d

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332565","number":332565,"commit":"6d550e2a4e58de08834f24153a2eefaf74f162f2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/332416","number":332416,"commit":"832ad4210b346abc9724585bdbeca03f70fe7e2e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/316260","number":316260,"commit":"ef53fc914eefde7b44514a483817fb36e725e2da"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314913","number":314913,"commit":"ed053abba60f7c332bad8a2c4424207fa254ede8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314817","number":314817,"commit":"846d6b41b830901d1f1c2e19ec9f3451feff8d13"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312148","number":312148,"commit":"faf050f745f7caa2a37433c9f1c81de50030c62d"}],"number":332565}
buildkite-pr-comment-->

PhilippeOberti reviewed (approved):
Looks awesome, thank you so much!

janmonschke reviewed (approved):
`kibana-cases` changes lgtm

PhilippeOberti commented:
@kapral18 hey, are you still planning on taking this over the finish line or you need some of our help here? I know it was close last time I reviewed it! Let us know how we can help :)

kapral18 commented:
> @kapral18 hey, are you still planning on taking this over the finish line or you need some of our help here? I know it was close last time I reviewed it! Let us know how we can help :)

Hey @PhilippeOberti sorry for this I was planning to get back to this but I didn't expect to get swamped by amount of high prio work I was introduced to ever since, so it was naive on my part to assume I could get back to this soon if ever.

That's why also I mentioned here https://github.com/elastic/kibana/pull/223838#issuecomment-3244826918 that you guys can take over to not wait on me.

If you still want me to handle this and have time to wait then wait, I can't promise when but as soon as I have an opening I will look into it. But Im most likely gonna just merge as is without finalizing the shared package I was planning because it's a lot of work and I don't have prio capacity for that at all.

So if you want it now you can take over merge conflicts and merge as IS. Let me know.

PhilippeOberti commented:
> > @kapral18 hey, are you still planning on taking this over the finish line or you need some of our help here? I know it was close last time I reviewed it! Let us know how we can help :)
> 
> Hey @PhilippeOberti sorry for this I was planning to get back to this but I didn't expect to get swamped by amount of high prio work I was introduced to ever since, so it was naive on my part to assume I could get back to this soon if ever.
> 
> That's why also I mentioned here [#223838 (comment)](https://github.com/elastic/kibana/pull/223838#issuecomment-3244826918) that you guys can take over to not wait on me.
> 
> If you still want me to handle this and have time to wait then wait, I can't promise when but as soon as I have an opening I will look into it. But Im most likely gonna just merge as is without finalizing the shared package I was planning because it's a lot of work and I don't have prio capacity for that at all.
> 
> So if you want it now you can take over merge conflicts and merge as IS. Let me know.

No rush on our end, I was just going through the list of PRs we have open, that's all 😄
